addon/bootstrap.js

```diff
@@ -62,14 +62,14 @@ async function startup(addonData, reason) {
   studyUtils.setVariation(variation);
   log.debug(`studyUtils has config and variation.name: ${variation.name}.  Ready to send telemetry`);
 
-
-  /** addon_install ONLY:
-    * - note first seen,
-    * - check eligible
-    */
-  if ((REASONS[reason]) === "ADDON_INSTALL") {
-    //  telemetry "enter" ONCE
+  /** addon_install and addon_upgrade (which is considered a new study) ONLY:
+   * - note first seen,
+   * - check eligible
+   */
+  if (reason === REASONS.ADDON_INSTALL || reason === REASONS.ADDON_UPGRADE) {
```

Comment:
I can see how this would be useful for testing, but what about in deployment? @motin, @gregglind , If we have already deployed an addon, and we have to patch it, is that an upgrade or a fresh install?

If we do upgrade addons after deployment, do we want the addon to completely reset? I.e. there would multiple `firstSeen` and other initial pings from the old version and the user might see an "install" UI element more than once.

This point is moot if we never see the `ADDON_UPGRADE` reason when the study is re-deployed through Normandy.

Reply:
Based on @mythmon's comments:
> The add-on manager calls something an upgrade when an add-on with the same ID as an existing add-on is installed, and a greater or equal version

> Normandy generally doesn't trigger upgrades, because Normandy doesn't change studies in practice. Instead it installs the add-on once, and then doesn't touch it until unenrollment time. In the future, when Normandy supports changing add-ons in place, it will trigger upgrades.

In light of this, my vote is that we future-proof the template and not equate `ADDON_INSTALL` with `ADDON_UPGRADE`. QA may have to remove the old version before re-installing a new version, but that seems a small price to pay to avoid potential unintended consequences of studies resetting on upgrade in the future. I think what happens on `ADDON_UPGRADE` will vary by study. @mythmon , @gregglind , @raymak, @motin , any feelings one way or the other?